Sorts/TimSort.java

```diff
@@ -0,0 +1,213 @@
+
+        int first_array_size = mid - start + 1, second_array_size = end - mid;
+        int array1[] = new int[first_array_size], array2[] = new int[second_array_size];
+        int i = 0, j = 0, k = 0;
+
+        // Building the two sub arrays from the array to merge later
+        for (i = 0; i < first_array_size; i++)
+            array1[i] = array[start + i];
+        for (i = 0; i < second_array_size; i++)
+            array2[i] = array[mid + 1 + i];
+
+        i = 0;
+        j = 0;
+        k = start;
+
+        while (i < first_array_size && j < second_array_size) {
+            if (array1[i] <= array2[j]) {
+                array[k] = array1[i];
+                i++;
+            } else {
+                array[k] = array2[j];
+                j++;
+            }
+            k++;
+        }
+
+        while (i < first_array_size) {
+            array[k] = array1[i];
+            k++;
+            i++;
+        }
+
+        while (j < second_array_size) {
+            array[k] = array2[j];
+            k++;
+            j++;
+        }
+    }
+
+    /**
+     * @brief Tim Sort Algorithm method.
+     */
+
+    public void algorithm() {
+        // Before Sorting
+        System.out.println("Before sorting the array: ");
+        this.showArrayElements();
+        System.out.println();
+
+        // Applying insertion sort on RUNS.
+        for (int i = 0; i < this.array_length; i += this.RUN)
+            this.insertion_sort(this.array, i, Math.min(i + this.RUN, (this.array_length - 1)));
+
+        for (int split = this.RUN; split < this.array_length; split = 2 * split) {
+            for (int start_idx = 0; start_idx < this.array_length; start_idx += 2 * split) {
+                int mid = start_idx + split - 1;
+                int end_idx = Math.min((start_idx + 2 * split - 1), (this.array_length - 1));
+
+                this.merge_runs(this.array, start_idx, mid, end_idx);
+            }
+        }
+        // After sorting
+        System.out.println("After sorting the array: ");
+        this.showArrayElements();
+        System.out.println();
+    }
+
+    /**
+     * @brief A method to show the elements inside the array.
+     */
+
+    public void showArrayElements() {
+        for (int i = 0; i < this.array.length; i++) {
+            System.out.print(this.array[i] + " ");
+        }
+        System.out.println();
+    }
+
+    /**
+     * @brief A method to test the sorting algorithm
+     */
+
+    static void test() {
+        int[] array = { 4, 1, 3, 17, 12, 11, 8 };
+        TimSort sorterObj1 = new TimSort();
+        TimSort sorterObj2 = new TimSort(50);
+        TimSort sorterObj3 = new TimSort(array);
+
+
+        sorterObj1.algorithm();
+        sorterObj2.algorithm();
+        sorterObj3.algorithm();
+
+        // Testing the first array
+        for(int i = 1; i < sorterObj1.array_length; i++) {
+            assert(!(sorterObj1.array[i] < sorterObj1.array[i - 1])) : "Array is not sorted";
+        }
```

Comment:
Simply you could have written this why turing around things to make it more complex
```suggestion
        for(int i = 0; i < sorterObj1.array_length - 1; i++) {
            assert((sorterObj1.array[i] < sorterObj1.array[i +1])) : "Array is not sorted";
        }
```